generate_network.py

```diff
@@ -0,0 +1,155 @@
+#! /usr/bin/env python3
+
+import matplotlib.pyplot as mp
+import networkx as nx
+import networkx.readwrite.json_graph
+import random, math
+import json
+import argparse,sys
+
+def write_json(filename, data_2_dump):
+    json.dump(data_2_dump, open(filename,'w'), indent=2)
+
+# has trouble with non-integer/non-hashable keys 
+def read_json(filename):
+    with open(filename) as f:
+        jdata = json.load(f)
+    return nx.node_link_graph(jdata)
+
+def draw(H):
+    nx.draw(H, pos=nx.kamada_kawai_layout(H), with_labels=True)
+    mp.show()
+    mp.savefig("topology.png", format="PNG")
+
+def init_arg_parser() :
+    # Initialize parser, add arguments and set the defaults
+    parser = argparse.ArgumentParser(
+            prog = 'generate_network',
+            description = '''Generates and outputs 
+            the Waku network conforming to input parameters''',
+            epilog = '''The defaults are: -o "Topology.json"; 
+            -n 1;  -t 1;  -p 1; -T "configuration_model"''')
+    parser.add_argument("-o", "--output", 
+            default='Topology.json', dest='fname', 
+            help='output json filename for the Waku network', 
+            type=str, metavar='<file_name>')
+    parser.add_argument("-n", "--numnodes", 
+            default=1, dest='num_nodes', 
+            help='number of nodes in the Waku network', 
+            type=int, metavar='<#nodes>')
+    parser.add_argument("-t", "--numtopics", 
+            default=1, dest='num_topics', 
+            help='number of topics in the Waku network', 
+            type=int, metavar='<#topics>')
+    parser.add_argument("-T", "--type", 
+            default="configuration_model", dest='nw_type', 
+            help='network type of the Waku network', 
+            type=str, metavar='<type>')
+    parser.add_argument("-p", "--numparts", 
+            default=1, dest='num_partitions', 
+            help='The number of partitions in the Waku network', 
+            type=int, metavar='<#partitions>')
+#    parser.add_argument("-e", "--numedges",
+#            default=1, dest='num_edges', 
+#            help='The number of edges in the Waku network', 
+#            type=int, metavar='#edges>')
+    return parser
+
+# https://networkx.org/documentation/stable/reference/generated/networkx.generators.degree_seq.configuration_model.html
+def generate_config_model(n):
```

Comment:
Not sure if by default I would use a random degree distribution as the graph generator. How about a Barabasi-Albert (althought I am not entirely sure if the network follows preferential attachment dynamics) model or a Watts-Strogatz instead

Reply:
we do not have enough real-world Waku nw data  to resolve it one way or another. the idea is to code more models and find out which one would be a best fit (mathematically or otherwise). I strongly suspect pub-sub nws will be assortative, but, yes, we need data.